fastestimator/op/tensorop/augmentation/cutmix_batch.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, Iterable, List, Tuple, TypeVar, Union
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+import torch
+
+from fastestimator.backend import random_mix_patch, roll
+from fastestimator.op.tensorop import TensorOp
+
+Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor)
+
+
+class CutMixBatch(TensorOp):
+    """This class perform cutmix augmentation on batch of tensors.
+
+    In this augmentation technique patches are cut and pasted among traning images where the ground truth labels are
+    also mixed proportionally to the area of the patches. This class should be used in conjunction with MixUpLoss to
+    perform CutMix training, which helps to reduce over-fitting, perform object detection, and against adversarial
+    attacks (https://arxiv.org/pdf/1905.04899.pdf)
+
+    Args:
+        inputs: key of the input to be cut-mixed
+        outputs: key to store the cut-mixed input
+        mode: what mode to execute in. Probably 'train'
+        alpha: the alpha value defining the beta distribution to be drawn from during training
+        framework: which framework the current Op will be executing in. Either 'tf' or 'torch'
+    """
+    def __init__(self,
+                 inputs: Union[str, List[str]],
+                 outputs: Union[str, List[str]],
```

Comment:
outputs has to be both lambda key and img_out key, so typing should only be List[str]. this should also be clarified in doc string and an assert in the __init

Reply:
or better yet, make it 'output' and 'mix_key' as separate args here, and default the 'mix_key' to 'lam' or something